src/REVDeployer.sol

```diff
@@ -941,6 +984,20 @@ contract REVDeployer is ERC2771Context, IREVDeployer, IJBRulesetDataHook, IJBCas
         (JBRulesetConfig[] memory rulesetConfigurations, bytes memory encodedConfiguration) =
             _makeRulesetConfigurations(configuration);
 
+        // Loop through all terminal configurations and check if the currency is the token being accepted.
+        // This both protects from misconfiguration and ensures the assumption(s) we make in `_makeLoanFundAccessLimits`
+        // are safe.
+        for (uint256 i; i < terminalConfigurations.length; i++) {
```

Comment:
if we're gonna do a loop, we might as well loop to derive `currency` based on loanSource.token?

i.e. if `terminalConfigurations[i].accountingContextsToAccept[j].token == loanSource.token` currency = terminalConfigurations[i].accountingContextsToAccept[j].currency